project-base/storefront/components/Blocks/Login/LoginForm.tsx

```diff
@@ -0,0 +1,133 @@
+import { yupResolver } from '@hookform/resolvers/yup';
+import { ExtendedNextLink } from 'components/Basic/ExtendedNextLink/ExtendedNextLink';
+import { WarningIcon } from 'components/Basic/Icon/WarningIcon';
+import { IconImage } from 'components/Basic/IconImage/IconImage';
+import { SubmitButton } from 'components/Forms/Button/SubmitButton';
+import { Form } from 'components/Forms/Form/Form';
+import { FormLine } from 'components/Forms/Lib/FormLine';
+import { PasswordInputControlled } from 'components/Forms/TextInput/PasswordInputControlled';
+import { TextInputControlled } from 'components/Forms/TextInput/TextInputControlled';
+import { useDomainConfig } from 'components/providers/DomainConfigProvider';
+import { GtmMessageOriginType } from 'gtm/enums/GtmMessageOriginType';
+import { Translate } from 'next-translate';
+import useTranslation from 'next-translate/useTranslation';
+import { FormProvider, SubmitHandler } from 'react-hook-form';
+import { usePersistStore } from 'store/usePersistStore';
+import { useLogin } from 'utils/auth/useLogin';
+import { blurInput } from 'utils/forms/blurInput';
+import { handleFormErrors } from 'utils/forms/handleFormErrors';
+import { useShopsysForm } from 'utils/forms/useShopsysForm';
+import { getInternationalizedStaticUrls } from 'utils/staticUrls/getInternationalizedStaticUrls';
+import * as Yup from 'yup';
+
+export type LoginFormProps = {
+    defaultEmail?: string;
+    shouldOverwriteCustomerUserCart?: boolean;
+};
+
+export const LoginForm: FC<LoginFormProps> = ({ defaultEmail, shouldOverwriteCustomerUserCart }) => {
+    const { t } = useTranslation();
+    const cartUuid = usePersistStore((store) => store.cartUuid);
+    const { url } = useDomainConfig();
+    const [resetPasswordUrl] = getInternationalizedStaticUrls(['/reset-password'], url);
+
+    const formProviderMethods = useShopsysForm(getLoginFormResolver(t), { email: defaultEmail ?? '', password: '' });
+    const login = useLogin();
+
+    const getSocialNetworkLoginUrl = (socialNetwork: 'facebook' | 'google' | 'seznam') =>
+        `/social-network/login/${socialNetwork}`;
+
+    const onLoginHandler: SubmitHandler<{ email: string; password: string }> = async (data) => {
```

Comment:
There should be some type of form.
By default it should be defined in project-base/storefront/types/form.ts